### PR TITLE
Adding indifferent access

### PR DESCRIPTION
**Pull request by elizasorber**
Tackling issue #1296 Added indifferent access to create_instances_from_response in the ShopifyAPI::Rest::Base class.

## Description

Fixes #<issue-number>

Please, include a summary of what the PR is for:
- What is the problem it is solving?
- What is the context of these changes?
- What is the impact of this PR?

## How has this been tested?

Please, describe the tests that you ran to verify your changes.

## Checklist:

- [ ] My commit message follow the pattern described in [here](https://chris.beams.io/posts/git-commit/)
- [x] I have performed a self-review of my own code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the project documentation.
- [ ] I have added a changelog line.

### CODE REVIEW

File: lib/shopify_api/rest/base.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # frozen_string_literal: true
 
 require "active_support/inflector"
+require 'active_support/all'
```
Comment (sle-c) [inline lib/shopify_api/rest/base.rb:5]: image

We should explicitly require hash_with_indifferent_access from active_support and not `all`

**Comment by sle-c** [comment]
@elizasorber , could you sign the CLA https://cla.shopify.com/ and comment "I have signed the CLA!"?

**Comment by elizasorber** [comment]
I have signed the CLA!

**Comment by sle-c** [comment]
hey @elizasorber , yeah we need a changelog entry and also change the import from `/all` to more explicit imports

**Comment by sle-c** [comment]
there's a rubocop rule that we need to fix as well as rebase on the main branch :)

**Review by lizkenyon — APPROVED** [review]

**Pull request merged**